### PR TITLE
Fix colorbar ticks not appearing in instrument view

### LINKED ISSUE #28142: No Tick Marks on Instrument View Colour Bar if Values too Close
### Expected behaviour
The scale on the workbench instrument viewer should always have some tick marks on it when using Log10. 
### Actual behaviour
If the max and min values are too close together no tick marks appear on the colour bar. 
### Steps to reproduce the behaviour
1. Load MUSR62250
2. Switch the colour bar to SymmetricLog10
3. See no tick marks
### Platforms affected
All, Workbench Only

### PR DESCRIPTION
**Pull request by PhilColebrooke**
**Description of work.**
This PR fixes a bug where the colorbar in the instrument view would sometimes have no ticks when the scale was set to SymmetricLog10.

image

**To test:**
1. Load MUSR62250
2. Switch the colorbar to SymmetricLog10
3. Check there are tick marks.

Fixes #28142 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/mplcpp/src/Colors.cpp [hunk 1]
```diff
@@ -176,8 +179,11 @@ Python::Object SymLogNorm::tickLocator() const {
   // Create log transform with base=10
   auto transform = scaleModule().attr("SymmetricalLogTransform")(
       10, Python::Object(pyobj().attr("linthresh")), m_linscale);
+  std::vector<float> subsVector(10);
+  std::iota(subsVector.begin(), subsVector.end(), 1);
```
Comment (martyngigg) [inline qt/widgets/mplcpp/src/Colors.cpp]: I think the MSVC build warnings are due to an integer `1` being use to initialize a `vector<float>`. Using `1.f` should do it.

File: qt/widgets/mplcpp/src/Colors.cpp [hunk 2]
```diff
@@ -176,8 +179,11 @@ Python::Object SymLogNorm::tickLocator() const {
   // Create log transform with base=10
   auto transform = scaleModule().attr("SymmetricalLogTransform")(
       10, Python::Object(pyobj().attr("linthresh")), m_linscale);
+  std::vector<float> subsVector(10);
```
Comment (martyngigg) [inline qt/widgets/mplcpp/src/Colors.cpp]: Could we add a small comment describing what this and the next line is doing? 
Particularly when using hard-coded numbers it's good to explain where they come from. It's also not quite clear that it's anything to do with the minor ticks until reading the matplotlib documentation.

**Review by martyngigg — DISMISSED** [review]

**Comment by DanielMurphy22** [comment]
Some of my thoughts: 
- How is it picking where the tick marks go. It seems that the ticks are at each 10,000 and the decimal assigned to it is just closeby. I think these are the right place but should maybe just say 3x10^4, 4x10^4 etc.
- It's a little worrying that there are 2 tick marks without a label on them! (Why do 3,4 and 6 x10^4 get labels but not 5 and 7 x10^4???)
- This is a much bigger job but should be be using `Log` rather than `SymmetricLog10`?

**Comment by PhilColebrooke** [comment]
I've changed the formatter so the tick label's are easier to understand now.
image

**Comment by martyngigg** [comment]
@DanielMurphy22 Could you take over reviewing this? I need to look at this plot on top thing.

**Review by DanielMurphy22 — APPROVED** [review]
To my points above:
**1)** Number format is now much improved!
**2)** Still odd that it will only
 display 2,3,4,6 but much clearer what the other non-labelled ticks actually would be!
Colorbarbetter
**3)** As I mentioned a much bigger job, so this close to a release I'm happy with this fix!

**Pull request merged**